### PR TITLE
support the new mongodb driver

### PR DESCRIPTION
**Pull request by tonicospinelli**
I know, it have many duplicated code from Mongo legacy. ASAP I will do a abstract driver to handle connection based on installed mongo extension :wink:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,23 +5,17 @@ php:
...
-  fast_finish: true
-  allow_failures:
-    - php: 7.0
-      env: MONGO_EXTENSION=mongo
-
 install:
-  - yes '' | pecl -q install -f ${MONGO_EXTENSION}-stable
-  - php --ri ${MONGO_EXTENSION} 
+  - pecl channel-update pecl.php.net
+  - yes '' | pecl -q install -f mongo-stable || exit 0
```
Comment (henriquemoody) [inline .travis.yml]: Is that necessary?

File: .scrutinizer.yml [hunk 2]
```diff
@@ -20,9 +20,11 @@ build:
     dependencies:
         before:
             - "yes '' | pecl -q install -f mongo-stable"
+            - "yes '' | pecl -q install -f mongodb-stable"
             - "sudo composer self-update"
         override:
-            - "composer install --no-dev --no-interaction --prefer-dist"
+            - composer update --no-interaction --prefer-dist
```
Comment (henriquemoody) [inline .scrutinizer.yml]: If you use [external code coverage](https://scrutinizer-ci.com/docs/tools/external-code-coverage/) no composer is needed here.

File: .travis.yml (hunk 1 above)
Comment (henriquemoody) [inline .travis.yml]: Oh, I got it now, I'm sorry.

Comment (tonicospinelli) [inline .travis.yml]: ok man! I forgive you :stuck_out_tongue:

File: .scrutinizer.yml (hunk 2 above)
Comment (tonicospinelli) [inline .scrutinizer.yml]: I'm running coverage on scrutinizer, look at these [lines](https://github.com/Respect/Structural/blob/master/.scrutinizer.yml#L26-L32)

**Pull request merged**